src/app/cpe/page/page.component.ts

```diff
@@ -0,0 +1,83 @@
+import {Component, Input, OnDestroy, OnInit} from '@angular/core';
+import {PageRepositoryService} from "../../repository/page-repository.service";
+import {Page, Project} from "../../repository/repository-model";
+import {Subscription} from "rxjs";
+import {ActivatedRoute} from "@angular/router";
+import {NavigationService} from "../../routing-module/navigation.service";
+
+@Component({
+  selector: 'app-page',
+  templateUrl: './page.component.html'
+})
+export class PageComponent implements OnInit, OnDestroy {
+
+  @Input() project?: Project;
+
+  pageId: string = '';
+
+  page: Page | undefined;
+
+  pageRouteSubscription: Subscription;
+
+  constructor(private pageService: PageRepositoryService,
+              private route: ActivatedRoute,
+              private navigationService: NavigationService) {
+    this.pageRouteSubscription = new Subscription();
+  }
+
+  ngOnInit(): void {
+    this.pageRouteSubscription = this.route.params.subscribe(parameter => {
+      if (parameter['pageId'] && parameter['pageId'] !== this.pageId) {
+        this.pageId = parameter['pageId'];
+        this.initPage(this.pageId);
+      }
+    });
+    if (!this.pageId) { // in case it is routed to the page component without a page id
+      this.navigateToDefaultPage();
+    }
+  }
+
+  /**
+   * initialized the page passed. Navigates to the default page, if no page id is passed at all or if the current project
+   * does not have a passed page.
+   * @param pageId: The id of the page to be loaded.
+   */
+  initPage(pageId: string | null) {
+    if ( pageId && this.project?.hasPage(pageId)) {
+      this.loadPage(pageId);
+    } else { // if there does not exist that page id for the project, we load the projects first page
+      console.warn(`No page with id ${pageId} found. Navigating to default/first page of project`)
+      this.navigateToDefaultPage();
+    }
+  }
+
+  /**
+   * loads the page passed. Navigates to the default page, if no page data can be retrieved from the api.
+   * @param pageId: The id of the page to be loaded.
+   */
+  loadPage(pageId: string) {
+    this.pageService.getPageByID(pageId).then(p => {
+      if (p) {
```

Comment:
This is a **fake api** with **mock components** for having any **routing** ready and develop a data driven **prove of concept**. We want to use https://github.com/Effect-TS/effect  anyways - so no fetch, no htttpclient get at all. There will be a NullPage of course. This will be removed anyways!  No need to blame a fake api with fake components. C'mon please ... That's like saying your services shouldn't be hardcoded when you mock them.